### DISCUSSION TITLE
Fullstack.Net - Integracao com o stripe - Teste de pagamento - 01:17 - Nao consigo acessar o stripe

### DISCUSSION THREAD

**Discussion by fabiorezendegithub** [balta-io/forum#720] [Tira Dúvidas] [answered]
balta , estou fazendo o curso de fullstack, parte de stripe, a aula testando o pagamento (01:17)

no momento que ele vai redirecionar para a tela do stripe, esta gerando o erro abaixo
![image](https://github.com/user-attachments/assets/263443ae-2398-4327-8789-539ed43b713e)

verificando no swagger, abusca pela sessao me parece estar normal
![image](https://github.com/user-attachments/assets/e171650a-f2f1-4e5c-a13d-b3dd4cd1f9d0)

nao consegui encontrar problemas no codigo(debug roda tudo normal, sem erros), entao nao sei se falta alguma configuracao no stripe ou no codigo, pode me dar uma ajuda?

**Comment by andrebaltieri** [comment, marked as answer]
Pelo que vi na imagem ele redirecionou certo para o Stripe, aparentemente teve algum erro de conexão... o erro ainda persiste?
Consegue testar de outra rede, tipo do 4G para garantir que não é nenhum bloqueio ai?

**Comment by fabiorezendegithub** [comment]
acho que nao balta, acabei de achar essa parte do stripe de logs, e parece que ta chegando certo la. Com relação a firewall, acho que não é não, to usando linux e deixei o firewall desativado.

![image](https://github.com/user-attachments/assets/1466bdd7-eefa-4e45-b33c-7417f0cc2543)

**Reply by andrebaltieri** [reply]
Que estranho... nada ainda? A única coisa que imagino que possa ser, seja o Token...
No Console do browser, quando você chega no Stripe, não mostra nda?

**Reply by fabiorezendegithub** [reply]
oi balta. então, descobri qual é o erro, mas não faço ideia do porque do erro rs.
veja so, quando fui olhar o console deu erro 401 nao autorized.
![image](https://github.com/user-attachments/assets/87123e77-a321-4835-aac3-357930fdfcdd)

ai fui la na parte de logs do stripe e copiei a url que o log retornava
![image](https://github.com/user-attachments/assets/c6f192f8-59c8-47dc-ad72-79555cd7c8ea)

e colei, e a pagina abriu certinho...
![image](https://github.com/user-attachments/assets/ddd96fe9-28b2-4b21-8d5c-3f7cf84be7c2)

ai fui comparar o url que peguei do log, com a url que tava sendo retornada, e tava diferente, por isso nao funcionava, acho eu...
![image](https://github.com/user-attachments/assets/fc7d0a76-c1a1-42b0-95cd-29e657176c01)


agora porque ta retornando errado, não faço idéia.

**Reply by andrebaltieri** [reply]
Estranho mesmo, ele tá gerando um Token inválido... tem certeza que sua chave do Stripe que está sendo utilizada é a mesma que você usou para gerar este Token? Consegue confirmar isto?

**Comment by RPegoli** [comment]
E aí pessoal, beleza? Cheguei aqui no mesmo erro. 
Também estou fazendo o curso de FullStack e estou seguindo à risca cada uma das aulas, e codificando junto com o Balta.
No meu caso eu percebi que Stripe está retornando um erro devido ao projeto estar rodando sobre o protocolo HTTP.
O Stripe retorna erro dizendo que para integrações com Stripe.Js é necssário utilizar HTTPS.
![2025-01-22_21h07_56](https://github.com/user-attachments/assets/decdbc1f-125c-4574-87b5-7bd23b6b8561)
Veja aí o erro que estoura quando executo o JSRuntime.
